tests/foreman/ui/test_jobinvocation.py

```diff
@@ -27,6 +30,114 @@ def module_rhel_client_by_ip(module_org, smart_proxy_location, rhel7_contenthost
     return rhel7_contenthost
 
 
+@pytest.mark.no_containers
+@pytest.mark.tier4
+def test_positive_bz2209968(
+    session,
+    module_org,
+    smart_proxy_location,
+    target_sat,
+):
+    """Check that full host group names are displayed when invoking a job
+
+    :id: 2301cd1d-ed82-4168-9f9b-d1661ac8fc5b
+
+    :steps:
+
+        1. Go to Monitor -> Jobs -> Run job
+        2. In "Target hosts and inputs" step, choose "Host groups" targeting
+
+    :expectedresults: Verify that in the dropdown, full hostgroup names are present, e.g. Parent/Child/Grandchild
+
+    :parametrized: yes
+
+    :customerscenario: true
+
+    :BZ: 2209968
+    """
+    names = valid_hostgroups_list_short()
+    parent1 = target_sat.cli_factory.hostgroup(
```

Comment:
i believe that if you constructed a nice (ordered) dictionary, that would describe the structure, you could generate the commands from a nice for loop.

something like this, perhaps:
```
In [27]: names = ['a','b','c','d','e','f','g','h']
    ...: tree = OrderedDict({
    ...: 'p1': {'n': names[0], 'p': None},
    ...: 'p2': {'n': names[1], 'p': None},
    ...: 'c1a': {'n': names[2], 'p': 'p1'},
    ...: 'c1b': {'n': names[3], 'p': 'p1'},
    ...: 'g1': {'n': names[4], 'p': 'c1a'},
    ...: 'g2': {'n': names[5], 'p': 'c1a'},
    ...: 'g3': {'n': names[6], 'p': 'c1b'},
    ...: 'g4': {'n': names[7], 'p': 'p2'},
    ...: })
```
..you could as well add 1 more key describing the expected name for later assertion

Reply:
Yes, that looks nicer